lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -128,11 +130,30 @@ def __init__(
         self.charm = charm
 
         super().__init__(charm, self.relation_name)
+        self.framework.observe(
+            charm.on[self.relation_name].relation_created, self._on_relation_created_event
+        )
         self.framework.observe(
             charm.on[self.relation_name].relation_changed, self._on_relation_changed
         )
         # TODO Future PRs handle scale down
 
+    def _on_relation_created_event(self, event):
+        """Sets database and extra user roles in the relation."""
+        if not self.charm.unit.is_leader():
+            return
+
+        if not self.charm.database:
+            logger.info("Waiting for database from application")
+            event.defer()
+            return
+
+        rel_data = {"database": self.charm.database}
+        if self.charm.extra_user_roles:
+            rel_data["extra-user-roles"] = str(self.charm.extra_user_roles)
```

Comment:
is it possible for `self.charm.extra_user_roles`  to have None as a value?

Reply:
yes! If its done, it gets defaulted to the [`default` role](https://github.com/canonical/mongodb-operator/blob/6faf6d83e9571cd4c4fa176eff60e83b8de9a987/lib/charms/mongodb/v1/mongodb_provider.py#L384)